cli/medperf/entities/dataset.py

```diff
@@ -26,7 +25,7 @@ class Dataset:
     data preparation output.
     """
 
-    def __init__(self, data_uid: int, ui: UI):
+    def __init__(self, data_uid: int, registration=None):
```

Comment:
what about expecting the `__init__` to always recieve a registration. `uid`-based retrievals would happen through `.get` or `.all`. Maybe follow this convention in all entities?

Reply:
So, the constructor of a Dataset will either receive the output hash (data_uid), or the registration. `Entity.get` will receive server uids only.

Do you think it would be better to split the constructor? something like abstract methods `Dataset.from_hash` and `Dataset.from_reg`.